### DISCUSSION TITLE
Experimenting with `EfficientAD` and other models!

### DISCUSSION THREAD

**Discussion by bebatrix** [open-edge-platform/anomalib#1519] [Q&A]
Hi everyone,

before I get to my question, I would like to say that I just came across this project and it looks absolutely fantastic! I'm relatively new to `PyTorch` and AI in general, and I can't say how happy I am that a library like this exists, so thank you so much :heart:!

Now, I just looked at the notebooks available to learn how to use this library, but I don't know if it will be suitable for my specific needs. I need to test several models for anomaly detection in a continuous learning scenario, so I'd like to be able to train/evaluate them manually and manipulate them via `torch`. Since I read in the `README.md` file in the root of this library that `EfficientAD` achieves very good results, I decided to start my learning of the library from that model. So, I basically created a new notebook where I simply tried to instantiate it as follows:
```python
from anomalib.models.efficient_ad import torch_model
import torch

device = "cuda:0"
dummy_input = torch.rand(3, 256, 256).to(device)

model = torch_model.EfficientAdModel(
    teacher_out_channels=512, input_size=(256, 256)
).to(device)

print(f"Output shape: {model(dummy_input).shape}")
```  

Unfortunately, I get this error:
```python
AttributeError: 'NoneType' object has no attribute 'device'
```

At this point I ask you, is it possible to train/evaluate the models manually, or is it absolutely necessary to use `PyTorch Lightning` and the scripts you provide?

Thank you very much and see you soon!

**Comment by abc-125** [comment]
Hi! PyTorch Lightning models contain a lot of crucial information for training, especially for EfficientAD. It handles the download of additional data for training, normalization of anomaly maps, etc. I think it makes more sense to work with PyTorch Lightning instead of just PyTorch, example can be found [here](https://github.com/openvinotoolkit/anomalib/blob/main/notebooks/200_models/201_fastflow.ipynb).

**Comment by samet-akcay** [comment]
Hi @bebatrix, glad you like the repo. You don't need the lightning. this is the exact reason why we created the `torch_model.py` for each model. If you would like to use it on your own, by all means, you could do. However, as @abc-125  pointed out, Lightning and anomalib handles so much of the boiler plate code that you will need to write on your own. 

Can you try instantiating the model safely first, then assign it to cuda device. Sth like:

```python
from anomalib.models.efficient_ad.torch_model import EfficientAdModel

model = EfficientAdModel(..)
```
See if the model is there. Then assign it to cuda to see why you get that error
